.github/workflows/publish.yml

```diff
@@ -85,6 +88,25 @@ jobs:
           chartpress --no-build
           ./tools/set-chart-yaml-annotations.py
 
+      - name: Publish images and chart with chartpress
+        if: >
+          github.repository == 'jupyterhub/zero-to-jupyterhub-k8s'
+          && github.event_name == 'push'
+        env:
+          GITHUB_REPOSITORY: "${{ github.repository }}"
+        run: |
           # Package the Helm chart and publish it to the gh-pages branch of
           # the jupyterhub/helm-chart repo.
           ./ci/publish
+
+      - name: Package helm chart as a CI artifact
+        if: github.event_name == 'pull_request'
+        run: helm package jupyterhub
+
+      # ref: https://github.com/actions/upload-artifact
+      - uses: actions/upload-artifact@v2
+        if: github.event_name == 'pull_request'
+        with:
+          name: jupyterhub-${{ github.sha }}
+          path: "jupyterhub-*.tgz"
+          if-no-files-found: error
```

Comment:
Does this have to be restricted to `pull_request` only? If it was for everything anyone who's enabled CI on their forks could get artifiacts built for their own branches

Reply:
Good point. I can make it exactly the inverse of the publish condition so any build that won't be published will get an artifact.

Reply:
Done.